### ISSUE TITLE
Build issue 

### ISSUE THREAD

**Issue by binoyp** [RUrlus/carma#60]
I am getting following cmake build out

```

Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  main.cpp
D:\ws\pybind\carma\include\carma/carma/converters.h(542,25): error C2882: 'pybind11': illegal use of namespace identifi
er in expression [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(542,25): error C4430: missing type specifier - int assumed. Note: C
++ does not support default-int [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(543,1): error C2146: syntax error: missing ';' before identifier 'N
AMESPACE_BEGIN' [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(546,61): error C2065: 'armaT': undeclared identifier [D:\ws\pybind\
build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(546,46): error C2923: 'carma::is_convertible': 'armaT' is not a val
id template type argument for parameter 'T' [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(546,69): error C2955: 'carma::is_convertible': use of class templat
e requires template argument list [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/utils.h(34): message : see declaration of 'carma::is_convertible' [D:\ws\pybind\
build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(546,77): error C2143: syntax error: missing ';' before '{' [D:\ws\p
ybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(546,77): error C2447: '{': missing function header (old-style forma
l list?) [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(631,15): error C2065: 'detail': undeclared identifier [D:\ws\pybind
\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(631,21): error C4430: missing type specifier - int assumed. Note: C
++ does not support default-int [D:\ws\pybind\build\cmake_example.vcxproj]
D:\ws\pybind\carma\include\carma/carma/converters.h(632,1): error C2146: syntax error: missing ';' before identifier 'N
AMESPACE_END' [D:\ws\pybind\build\cmake_example.vcxproj]`
```
Cmake setup is as follows
```

CMAKE_MINIMUM_REQUIRED(VERSION 3.7)
project(cmake_example)


set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

add_subdirectory(pybind11)
# add_subdirectory(carma)
include_directories(carma/third_party/armadillo-code/include)
include_directories(carma/include)
link_directories("lib")
pybind11_add_module(cmake_example main.cpp)




```
my lib directories contains blas libs required for armadillo

What's going wrong here?

**Comment by RUrlus** [comment]
Have a look at the [examples/CMakeLists.txt](https://github.com/RUrlus/carma/blob/master/examples/CMakeLists.txt) for an example on how include CARMA using CMake. I am not entirely sure what is causing the errors but I think it's a linking issue with your example CMake.

**Comment by RUrlus** [comment]
@binoyp any luck with the example CMakeLists?

**Comment by binoyp** [comment]
Thanks. Fixed it. It was a mistake in CMakeLists.txt

**Comment by binoyp** [comment]
It is due to something else. I added carma, armadillo and pybind11 as git submodules. It worked in my personal computer.

But when i cloned in my office PC same error came. Environment variables are same. I have no clue what's is causing this?

**Comment by binoyp** [comment]
Figured out the issue. 

Macro `NAMESPACE_BEGIN` has been changed to `PYBIND11_NAMESPACE_BEGIN` and `NAMESPACE_END` to 
`PYBIND11_NAMESPACE_END`

For details 

https://github.com/pybind/pybind11/blob/37f845a1dc171d981149a53f01c13c6ba85e0f2e/include/pybind11/detail/common.h#L16

**Comment by RUrlus** [comment]
Thanks for letting me know! Fixed in 9006c6cc0d41ed27c3ab0e23a2ef4b196e3ca8af
I'll leave the issue open till the branch is merged in, for now reverting to pybind11 2.4.3 is a quick fix
